pkg/cache/index/index_test.go

```diff
@@ -110,68 +108,68 @@ func TestReap(t *testing.T) {
 	idx.UpdateObject(&Object{Key: "cache.index", Value: []byte("test_value")})
 
 	// add expired key to cover the case that the reaper remove it
-	idx.UpdateObject(&Object{Key: "test.1", Value: []byte("test_value"), Expiration: time.Now().Add(-time.Minute)})
+	idx.UpdateObject(&Object{Key: "test.1", Value: []byte("test_value"), Expiration: atomicx.NewTime(time.Now().Add(-time.Minute))})
 
 	// add key with no expiration which should not be reaped
 	idx.UpdateObject(&Object{Key: "test.2", Value: []byte("test_value")})
 
 	// add key with future expiration which should not be reaped
-	idx.UpdateObject(&Object{Key: "test.3", Value: []byte("test_value"), Expiration: time.Now().Add(time.Minute)})
+	idx.UpdateObject(&Object{Key: "test.3", Value: []byte("test_value"), Expiration: atomicx.NewTime(time.Now().Add(time.Minute))})
 
 	// trigger a reap that will only remove expired elements but not size down the full cache
 	idx.reap()
 
 	// add key with future expiration which should not be reaped
-	idx.UpdateObject(&Object{Key: "test.4", Value: []byte("test_value"), Expiration: time.Now().Add(time.Minute)})
+	idx.UpdateObject(&Object{Key: "test.4", Value: []byte("test_value"), Expiration: atomicx.NewTime(time.Now().Add(time.Minute))})
 
 	// add key with future expiration which should not be reaped
-	idx.UpdateObject(&Object{Key: "test.5", Value: []byte("test_value"), Expiration: time.Now().Add(time.Minute)})
+	idx.UpdateObject(&Object{Key: "test.5", Value: []byte("test_value"), Expiration: atomicx.NewTime(time.Now().Add(time.Minute))})
 
 	// add key with future expiration which should not be reaped
-	idx.UpdateObject(&Object{Key: "test.6", Value: []byte("test_value"), Expiration: time.Now().Add(time.Minute)})
+	idx.UpdateObject(&Object{Key: "test.6", Value: []byte("test_value"), Expiration: atomicx.NewTime(time.Now().Add(time.Minute))})
 
 	// trigger size-based reap eviction of some elements
 	idx.reap()
 
-	if _, ok := idx.Objects["test.1"]; ok {
+	if _, ok := idx.Objects.Load("test.1"); ok {
 		t.Errorf("expected key %s to be missing", "test.1")
 	}
 
-	if _, ok := idx.Objects["test.2"]; ok {
+	if _, ok := idx.Objects.Load("test.2"); ok {
 		t.Errorf("expected key %s to be missing", "test.2")
 	}
 
-	if _, ok := idx.Objects["test.3"]; ok {
+	if _, ok := idx.Objects.Load("test.3"); !ok {
```

Comment:
behavior appears to be a littler different, investigating why still

Reply:
adding some debugging locally, this is what I see in this function:
```
=== RUN   TestReap
    index_test.go:120: objects before reap 4
    index_test.go:122: objects after reap 2
    index_test.go:134: objects before reap 5
    index_test.go:136: objects after reap 5
    index_test.go:168: objects before reap 6
    index_test.go:170: objects after reap 1
```
vs similar changes to main
```
=== RUN   TestReap
    index_test.go:121: objects before reap 4
    index_test.go:124: objects after reap 3
    index_test.go:136: objects before reap 6
    index_test.go:138: objects after reap 2
    index_test.go:170: objects before reap 3
    index_test.go:172: objects after reap 1
``

Reply:
the initial reap is definitely a bug, some of the others seem ok. continuing to debug. May be something to do with the change in zero time value.

Reply:
The issue is definitely with the atomicx.Time zero value; going to rework that